jax/_src/checkify.py

```diff
@@ -796,13 +804,8 @@ def add_nan_check(prim):
 add_nan_check(lax.add_p)
 add_nan_check(lax.sub_p)
 add_nan_check(lax.convert_element_type_p)
-add_nan_check(lax.broadcast_in_dim_p)
```

Comment:
Why do we need to remove these checks? Won't KeyArrays get caught by the `isnan` check above?

Reply:
agreed, I figured they don't produce nans anyway, so might be able to remove them (see PR description), but I'll leave it for a follow-up change which reviews the set (eg. are we missing any primitives).

Reverted the removals for now.

Reply:
Oh I see. Yea these can be removed in a follow up PR.